webtau-core/src/main/java/com/twosigma/webtau/documentation/DocumentationArtifacts.java

```diff
@@ -16,13 +16,27 @@
 
 package com.twosigma.webtau.documentation;
 
+import com.twosigma.webtau.data.table.Record;
+import com.twosigma.webtau.data.table.TableData;
+import com.twosigma.webtau.utils.CsvUtils;
 import com.twosigma.webtau.utils.FileUtils;
+import com.twosigma.webtau.utils.JsonUtils;
 
 import java.nio.file.Path;
 
 public class DocumentationArtifacts {
-    public static void create(Class testClass, String artifactName, String textContent) {
+    public static void create(Class<?> testClass, String artifactName, String text) {
         Path path = DocumentationArtifactsLocation.classBasedLocation(testClass).resolve(artifactName);
-        FileUtils.writeTextContent(path, textContent);
+        FileUtils.writeTextContent(path, text);
+    }
+
+    public static void createAsJson(Class<?> testClass, String artifactName, TableData tableData) {
```

Comment:
Do we not want overrides for something other than table data?

Reply:
Don't have usecases yet. There is a `String`  fallback